sed/binning.py

```diff
@@ -364,9 +442,9 @@ def _hist_from_bin_ranges(
 
 def numba_histogramdd(
     sample: np.array,
-    bins: Sequence,
-    ranges: Sequence,
-) -> Tuple[np.array, np.array]:
+    bins: Union[Sequence[Union[int, np.ndarray]], np.ndarray],
+    ranges: Sequence = None,
+) -> Tuple[np.ndarray, np.ndarray]:
     """Wrapper for the Number pre-compiled binning functions.
```

Comment:
I would skip the reason why the function was created. The user do not care :)
Regarding the overflow, I would use an assert for that

Reply:
good point! I will look into the function descriptions again